.github/workflows/publish.yml

```diff
@@ -1,52 +1,57 @@
 name: Release
 
 on:
-  # NOTE: PR trigger is to ensure changes do not break packaging.
-  pull_request:
-  release:
-    types: [released]
-
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    workflow_dispatch:
+    pull_request:
+    release:
+        types: [created]
 
 jobs:
-  build-n-publish:
-    name: Build and publish Python distributions to PyPI
-    runs-on: ubuntu-latest
-    permissions:
-      id-token: write  # IMPORTANT: this permission is mandatory for trusted publishing
-    if: github.repository == 'phoebe-project/phoebe2'
-
-    steps:
-    - uses: actions/checkout@v4
-      with:
-        fetch-depth: 0
-
-    - uses: actions/setup-python@v5
-      with:
-        python-version: '3.10'
-
-    - name: Install python-build and twine
-      run: python -m pip install build "twine>=3.3"
-
-    - name: Build package
-      run: python -m build --sdist --wheel .
-
-    - name: List result
-      run: ls -l dist
-
-    - name: Check dist
-      run: python -m twine check --strict dist/*
-
-    - name: Test package
-      run: |
-        cd ..
-        python -m venv testenv
-        testenv/bin/pip install pytest phoebe2/dist/*.whl
-        testenv/bin/python -c "import phoebe"
-
-    # NOTE: Do not run this part for PR testing.
-    - name: Publish distribution to PyPI
-      uses: pypa/gh-action-pypi-publish@release/v1
-      if: github.event_name != 'pull_request'
\ No newline at end of file
+    build-sdist:
+        name: Package source distribution
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - name: Build sdist tarball
+              run: |
+                pipx run build --sdist
+            - uses: actions/upload-artifact@v4
+              with:
+                name: cibw-sdist
+                path: dist/*.tar.gz
+
+    build-wheels:
+        name: Build wheels on ${{ matrix.os }}
+        runs-on: ${{ matrix.os }}
+        strategy:
+            matrix:
+                os: [ubuntu-22.04, macos-13, macos-14]
```

Comment:
As of Sep 2024, ubuntu-24.04 runner is supported; can you change it and see if that fixes the issue?

Reply:
nope, same error 🤷‍♂️